code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/github/GithubTokenCreator.java

```diff
@@ -35,59 +19,15 @@
     @Inject
     GithubClient githubClient;
-    @Inject
-    private TokenService tokenService;
 
     public Token getGithubToken(String accessToken) {
         if (!isConfigured()) {
             throw new ClientVisibleException(ResponseCodes.SERVICE_UNAVAILABLE, GithubConstants.CONFIG, "No Github Client id and secret found.", null);
         }
-        List<TeamAccountInfo> teamsAccountInfo = new ArrayList<>();
-        GithubAccountInfo userAccountInfo = githubClient.getUserAccountInfo(accessToken);
-        List<GithubAccountInfo> orgAccountInfo = githubClient.getOrgAccountInfo(accessToken);
-        Set<Identity> identities = new HashSet<>();
-
-        Identity user = userAccountInfo.toIdentity(GithubConstants.USER_SCOPE);
-        identities.add(user);
-
-        for (GithubAccountInfo info : orgAccountInfo) {
-            teamsAccountInfo.addAll(githubClient.getOrgTeamInfo(accessToken, info.getAccountName()));
-            Identity org = info.toIdentity(GithubConstants.ORG_SCOPE);
-            identities.add(org);
-        }
-
-        for (TeamAccountInfo info : teamsAccountInfo) {
-            Identity team = new Identity(GithubConstants.TEAM_SCOPE, info.getId(), info.getOrg() + ":" + info.getName(),
-                    null, null, null);
-            identities.add(team);
-        }
-        List<String> idList = githubUtils.identitiesToIdList(identities);
-        Account account = githubUtils.getOrCreateAccount(user, identities, null, true);
-        if (account == null){
-            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "FailedToGetAccount");
-        }
-        Map<String, Object> jsonData = new HashMap<>();
-        jsonData.put(TokenUtils.TOKEN, GithubConstants.GITHUB_JWT);
-        jsonData.put(TokenUtils.ACCOUNT_ID, userAccountInfo.getAccountId());
-        jsonData.put(GithubConstants.USERNAME, userAccountInfo.getAccountName());
-        jsonData.put(TokenUtils.ID_LIST, idList);
-        DataAccessor.fields(account).withKey(GithubConstants.GITHUB_ACCESS_TOKEN).set(accessToken);
-        objectManager.persist(account);
-        account = objectManager.reload(account);
-        String accountId = (String) ApiContext.getContext().getIdFormatter().formatId(objectManager.getType(Account.class), account.getId());
-        Date expiry = new Date(System.currentTimeMillis() + SecurityConstants.TOKEN_EXPIRY_MILLIS.get());
-        String jwt = tokenService.generateEncryptedToken(jsonData, expiry);
-        //LEGACY: Used for old Implementation of projects/ Identities. Remove when vincent changes to new api.
-        return new Token(jwt, user.getName(), null, teamsAccountInfo, SecurityConstants.SECURITY.get(),
-                GithubConstants.GITHUB_CLIENT_ID.get(), account.getKind(), SecurityConstants.AUTH_PROVIDER.get(),
-                accountId, new ArrayList<>(identities), user);
+        return githubUtils.createToken(githubClient.getIdentities(accessToken), null);
```

Comment:
This logic was half repeated from other classes as a result it was moved to tokenUtils and getIdentities in the githubClient.